PlagiarismPlugin.inc.php

```diff
@@ -43,172 +75,431 @@ public function getDescription() {
 		return parent::getEnabled($contextId) || Config::getVar('ithenticate', 'ithenticate');
 	}
 
+	/**
+	 * Add the IThenticate EULA url as a checklist of submission process
+	 * 
+	 * @param string $hookName `TemplateManager::display` or `TemplateManager::fetch`
+	 * @param array $params
+	 * 
+	 * @return bool
+	 */
+	public function addEulaToChecklist($hookName, $params) {
+		$templateManager = & $params[0]; /** @var TemplateManager $templateManager */
+		$context = $templateManager->getTemplateVars('currentContext'); /** @var Context $context */
+		
+		if (!$context || strtolower($templateManager->getTemplateVars('requestedPage') ?? '') !== 'submission') {
+			return false;
+		}
+
+		$eualDetails = $this->getContextEulaDetails($context);
+		
+		foreach($context->getData('submissionChecklist') as $locale => $checklist) {
+			array_push($checklist, [
+				'order' => (collect($checklist)->pluck('order')->sort(SORT_REGULAR)->last() ?? 0) + 1,
+				'content' => __('plugins.generic.plagiarism.submission.checklist.eula', [
+					'localizedEulaUrl' => $eualDetails[$locale]['url']
+				]),
+			]);
+
+			$context->setData('submissionChecklist', $checklist, $locale);
+		}
+
+		$templateManager->assign('currentContext', $context);
+		
+		return false;
+	}
+
+	/**
+	 * Add properties for this type of public identifier to the submission entity's list for
+	 * storage in the database.
+	 * 
+	 * @param string $hookName `Schema::get::submission`
+	 * @param array $params
+	 * 
+	 * @return bool
+	 */
+	public function addPlagiarismCheckDataToSubmissionSchema($hookName, $params) {
+		$schema =& $params[0];
+
+		$schema->properties->ithenticate_id = (object) [
+			'type' => 'string',
+			'description' => 'The json encoded data map of submission file id to ithenticate submission id for each file submission file',
+			'apiSummary' => true,
+			'validation' => ['nullable'],
+		];
+
+		$schema->properties->ithenticate_eula_version = (object) [
+			'type' => 'string',
+			'description' => 'The iThenticate EULA version which has been agreed at submission checklist',
+			'apiSummary' => true,
+			'validation' => ['nullable'],
+		];
+
+		$schema->properties->ithenticate_eula_url = (object) [
+			'type' => 'string',
+			'description' => 'The iThenticate EULA url which has been agreen at submission checklist',
+			'apiSummary' => true,
+			'validation' => ['nullable'],
+		];
+
+		return false;
+	}
+
+	/**
+	 * Add properties for this type of public identifier to the context entity's list for
+	 * storage in the database.
+	 * 
+	 * @param string $hookName `Schema::get::context`
+	 * @param array $params
+	 * 
+	 * @return bool
+	 */
+	public function addPlagiarismCheckWebhookDataToSchema($hookName, $params) {
+		$schema =& $params[0];
+
+		$schema->properties->ithenticate_webhook_signing_secret = (object) [
+			'type' => 'string',
+			'description' => 'The iThenticate service webook registration signing secret',
+			'apiSummary' => true,
```

Comment:
I'm pretty sure we don't want this to be added to API GET responses, either for the summary or the full GET! Make the signing secret `writeOnly` and exclude it from the `apiSummary` (and I guess also the webhook ID).